src/video_stream.cpp

```diff
@@ -136,14 +136,16 @@ virtual void do_capture() {
         {
             camera_fps_rate.sleep();
         }
+        NODELET_INFO_DEBUG("Current frame_counter: " << frame_counter << " latest_config.stop_frame - latest_config.start_frame: " << latest_config.stop_frame - latest_config.start_frame);
         if (video_stream_provider_type == "videofile" &&
-            frame_counter == latest_config.stop_frame - latest_config.start_frame)
+            frame_counter >= latest_config.stop_frame - latest_config.start_frame)
```

Comment:
There are cases where the `frame_counter` variable can go over this check and then we are stuck forever on the `Could not capture frame` error. By doing `>=` we make sure to reset.